File: packages/compose/src/main/kotlin/ai/mobileagentos/compose/readiness/ReadinessDimension.kt
package ai.mobileagentos.compose.readiness

enum class ReadinessDimension {
    CONTEXT_AWARENESS,
    MEMORY_QUALITY,
    PRIVACY_SAFETY,
    ACTION_RELIABILITY,
    OFFLINE_RESILIENCE,
    UX_CONTINUITY,
    APP_STORE_READINESS
}

data class DimensionScore(
    val score: Int,
    val notes: String? = null
)

File: packages/swift/Sources/MobileAgentOS/Readiness/ReadinessDimension.swift
import Foundation

/// Dimensions of mobile AI readiness
public enum ReadinessDimension: String, Codable, CaseIterable {
    case contextAwareness
    case memoryQuality
    case privacySafety
    case actionReliability
    case offlineResilience
    case uxContinuity
    case appStoreReadiness
    
    public var weight: Double {
        switch self {
        case .contextAwareness: return 0.15
        case .memoryQuality: return 0.15
        case .privacySafety: return 0.20
        case .actionReliability: return 0.20
        case .offlineResilience: return 0.10
        case .uxContinuity: return 0.10
        case .appStoreReadiness: return 0.10
        }
    }
}